shopinvader_delivery_carrier/services/delivery.py

```diff
@@ -0,0 +1,117 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 ACSONE SA/NV (<http://acsone.eu>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import fields
+from odoo.addons.base_rest.components.service import to_int
+from odoo.addons.component.core import Component
+
+
+class AbstractSaleService(Component):
+    _inherit = "base.shopinvader.service"
+    _name = "shopinvader.delivery.service"
+    _usage = "delivery"
+    _expose_model = "stock.picking"
+
+    def search(self, **params):
+        """
+        Get every delivery related to logged user
+        :param params: dict/json
+        :return: dict
+        """
+        return self._paginate_search(**params)
+
+    def _validator_search(self):
+        """
+        Validator for the search
+        :return: dict
+        """
+        schema = {
+            "per_page": {
+                "coerce": to_int,
+                "nullable": True,
+                "type": "integer",
+            },
+            "page": {"coerce": to_int, "nullable": True, "type": "integer"},
+        }
+        return schema
+
+    def _get_parser_picking(self):
+        """
+        Get the parser of stock.picking
+        :return: list
+        """
+        to_parse = [
+            "id:delivery_id",
+            "carrier_tracking_ref:tracking_reference",
+            ("carrier_id:carrier", ("name",)),
+        ]
+        return to_parse
+
+    def _get_parser_sale_order(self):
```

Comment:
@acsonefho For new services we should start to provide the schema for the responses..